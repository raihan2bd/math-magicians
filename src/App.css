* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.App {
  text-align: center;
}

.App-header {
  background-color: #3f3f3f;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
