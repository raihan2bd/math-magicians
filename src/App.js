import React from 'react';
import { Route, Routes } from 'react-router-dom';

import Layout from './components/UI/Layout';
import HomePage from './pages/HomePage';
import QuotePage from './pages/QuotePage';
import CalculatorPage from './pages/CalculatorPage';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Layout>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/calculator" element={<CalculatorPage />} />
          <Route path="/quote" element={<QuotePage />} />
        </Routes>
      </Layout>
    );
  }
}

export default App;
