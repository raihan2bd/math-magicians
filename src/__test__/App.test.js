import { render, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import App from '../App';

describe('testing App.js with snapshot', () => {
  test('test App.js dom snapshot', () => {
    const app = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();

    expect(app).toMatchSnapshot();
  });

  test('testing home routes', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    expect(screen.getByText('Home')).toBeInTheDocument();

    expect(screen.getByText('Welcome to our page!')).toBeInTheDocument();
    expect(screen.queryByText('Lets do some Math!')).toBeNull();
    expect(screen.queryByText('Refresh')).toBeNull();
  });

  test('test calculator page', () => {
    render(
      <MemoryRouter initialEntries={['/calculator']}>
        <App />
      </MemoryRouter>,
    );

    expect(
      screen.getByRole('heading', { name: 'Lets do some Math!' }),
    ).toBeInTheDocument();
  });

  test('test quote page', () => {
    render(
      <MemoryRouter initialEntries={['/quote']}>
        <App />
      </MemoryRouter>,
    );

    expect(screen.getByRole('button', { name: 'Refresh' })).toBeInTheDocument();
  });
});
