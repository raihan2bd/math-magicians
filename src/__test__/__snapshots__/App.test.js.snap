// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App.js with snapshot test App.js dom snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <nav
      className="nav"
    >
      <div
        className="nav_brand"
      >
        <h1>
          <a
            href="/"
            onClick={[Function]}
          >
            Math Magicians
          </a>
        </h1>
      </div>
      <ul
        className="nav_group"
      >
        <li
          className="nav_item"
        >
          <a
            aria-current="page"
            className="nav_link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="nav_link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            className="nav_link"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="main"
  >
    <div
      className="container"
    >
      <section
        className="hompage_section"
      >
        <h2>
          Welcome to our page!
        </h2>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Quas quis harum, quibusdam ab distinctio praesentium voluptatibus alias molestiae possimus eveniet nulla repudiandae minima corrupti quo non provident perspiciatis voluptatem at. Ratione, sit? Quam itaque vero esse, nesciunt inventore doloremque ratione ducimus non voluptatem voluptates quae illo veritatis! Illo minima rerum, illum mollitia explicabo iste, hic consequatur ipsum distinctio impedit vero! Nesciunt vel in molestias qui temporibus consectetur perferendis recusandae quos non vero. Temporibus unde similique neque maxime vero atque, rerum culpa non vel fugit, quis perferendis, eum quas repellendus voluptatem!
        </p>
      </section>
    </div>
  </main>,
]
`;
